components/motor/gpio/encoded_test.go

```diff
@@ -186,31 +188,45 @@ func TestEncodedMotor(t *testing.T) {
 	})
 
 	t.Run("encoded motor test GoFor forward", func(t *testing.T) {
-		t.Skip("temporary skip for flake")
-		test.That(t, m.goForInternal(10, 1, 1), test.ShouldBeNil)
-		testutils.WaitForAssertion(t, func(tb testing.TB) {
-			tb.Helper()
-			on, powerPct, err := m.IsPowered(context.Background(), nil)
-			test.That(tb, on, test.ShouldBeTrue)
-			test.That(tb, powerPct, test.ShouldBeGreaterThan, 0)
-			test.That(tb, err, test.ShouldBeNil)
-		})
+		initpos, err := m.Position(context.Background(), nil)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, m.GoFor(context.Background(), 10, 1, nil), test.ShouldBeNil)
+
+		finalpos, err := m.Position(context.Background(), nil)
+		test.That(t, err, test.ShouldBeNil)
+
+		test.That(t, initpos < finalpos, test.ShouldBeTrue)
+
+		// testutils.WaitForAssertion(t, func(tb testing.TB) {
```

Comment:
I could never get these to pass, the lines before blocked until stopped, and then stop reported a stopped motor state.